File: cside/src/data/Identifier.cpp
//
// Created by TheAviary on 4/7/2023.
//
#include "Identifier.h"

Identifier::Identifier(string pack, string name) {
    this->pack = pack;
    this->name = name;
}

Identifier::Identifier(string name) {
    this->pack = DEFAULT_PACK;
    this->name = name;
}

File: core/src/main/kotlin/org/nullversionnova/common/Identifier.kt
package org.nullversionnova.common

import org.nullversionnova.server.Engine

data class Identifier(var pack: String = Engine.pack_identifier, var name: String = "default") {
    constructor(identifier: String) : this(identifier.substringBefore(':'), identifier.substringAfter(':'))
    override fun toString() : String { return "$pack:$name" }
}
